components/script/dom/element.rs

```diff
@@ -692,6 +693,29 @@ impl LayoutElementHelpers for LayoutJS<Element> {
         }
     }
 
+    #[allow(unsafe_code)]
+    fn get_lang_for_layout(&self) -> String {
+        unsafe {
+            let mut current_node = Some(self.upcast::<Node>());
+            while let Some(node) = current_node {
+                current_node = node.parent_node_ref();
+                match node.downcast::<Element>().map(|el| el.unsafe_get()) {
+                    Some(elem) => {
+                        if let Some(attr) = (*elem).get_attr_val_for_layout(&ns!(xml), &local_name!("lang")) {
+                            return attr.to_owned();
+                        } else if let Some(attr) = (*elem).get_attr_val_for_layout(&ns!(), &local_name!("lang")) {
+                            return attr.to_owned();
+                        }
+                    }
+                    None => continue
```

Comment:
I'm not sure if it's possible to have a non-elemnt node be a parent of an element. However your solution doesn't quite work, because we need some way of accessing the parent element of `current_element`, and there's no convenience method for this yet, meaning we'll need to upcast it back to a Node struct anyway just to call `parent_node_ref`.